sdk/src/azure/core/az_http_request.c

```diff
@@ -67,27 +69,33 @@ AZ_NODISCARD az_result az_http_request_append_path(az_http_request* ref_request,
                                                : ref_request->_internal.url_length;
 
   /* use replace twice. Yes, we will have 2 right shift (one on each replace), but we rely on
-   * replace functionfor doing this movements only and avoid updating manually. We could also create
-   * a temp buffer to join "/" and path and then use replace. But that will cost us more stack
-   * memory.
+   * replace function for doing this movements only and avoid updating manually. We could also
+   * create a temp buffer to join "/" and path and then use replace. But that will cost us more
+   * stack memory.
    */
   AZ_RETURN_IF_FAILED(_az_span_replace(
       ref_request->_internal.url,
       ref_request->_internal.url_length,
       query_start,
       query_start,
-      AZ_SPAN_FROM_STR("/")));
+      AZ_SPAN_FROM_STR("/"),
+      false,
+      NULL));
   query_start += 1; // a size of "/"
   ++ref_request->_internal.url_length;
 
+  int32_t path_size = az_span_size(path);
   AZ_RETURN_IF_FAILED(_az_span_replace(
       ref_request->_internal.url,
       ref_request->_internal.url_length,
       query_start,
       query_start,
-      path));
-  query_start += az_span_size(path);
-  ref_request->_internal.url_length += az_span_size(path);
+      path,
+      !is_url_encoded,
```

Comment:
I don't think replace should follow same approach as append_path/set_query.

For replace, since it is core and private function, we want to control the future of the input parameters and how the replace should be done. Replace doesn't not care about url-encoding, so it just ask about doing or not.
For append_path we just want to describe how the input is passed. This functions (append/query) must ensure the url-encoding, so it ask hows the input to make sure it keeps url-encoded